File: src/main/java/com/eventmanagementsystem/service/OrganizerService.java
package com.eventmanagementsystem.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.eventmanagementsystem.domain.Organizer;
import com.eventmanagementsystem.dto.OrganizerDTO;
import com.eventmanagementsystem.dto.OrganizerSearchDTO;
import com.eventmanagementsystem.dto.OrganizerPageDTO;
import com.eventmanagementsystem.dto.OrganizerConvertCriteriaDTO;
import com.eventmanagementsystem.service.GenericService;
import com.eventmanagementsystem.dto.common.RequestDTO;
import com.eventmanagementsystem.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface OrganizerService extends GenericService<Organizer, Integer> {

	List<Organizer> findAll();

	ResultDTO addOrganizer(OrganizerDTO organizerDTO, RequestDTO requestDTO);

	ResultDTO updateOrganizer(OrganizerDTO organizerDTO, RequestDTO requestDTO);

    Page<Organizer> getAllOrganizers(Pageable pageable);

    Page<Organizer> getAllOrganizers(Specification<Organizer> spec, Pageable pageable);

	ResponseEntity<OrganizerPageDTO> getOrganizers(OrganizerSearchDTO organizerSearchDTO);
	
	List<OrganizerDTO> convertOrganizersToOrganizerDTOs(List<Organizer> organizers, OrganizerConvertCriteriaDTO convertCriteria);

	OrganizerDTO getOrganizerDTOById(Integer organizerId);







}

File: eventmanagementsystem_ui/src/services/OrganizerService.js
import http from "../http-common";

class OrganizerService {
  getAllOrganizers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/organizer/organizers`, searchDTO);
  }

  get(organizerId) {
    return this.getRequest(`/organizer/${organizerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/organizer?field=${matchData}`, null);
  }

  addOrganizer(data) {
    return http.post("/organizer/addOrganizer", data);
  }

  update(data) {
  	return http.post("/organizer/updateOrganizer", data);
  }
  
  uploadImage(data,organizerId) {
  	return http.postForm("/organizer/uploadImage/"+organizerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new OrganizerService();
